pages/scenarios/[id].vue

```diff
@@ -96,13 +150,63 @@ import type { Parameter } from "~/types/parameterTypes";
 
 const appStore = useAppStore();
 
+const openedTimeSeriesAccordions = ref<string[]>([]);
+const hideTimeSeriesTooltips = ref(false);
+
 const paramDisplayText = (param: Parameter) => {
-  const rawVal = appStore.currentScenario!.parameters[param.id].toString();
-  return param.options ? param.options.find(({ id }) => id === rawVal)!.label : rawVal;
+  if (appStore.currentScenario.parameters && appStore.currentScenario.parameters[param.id]) {
+    const rawVal = appStore.currentScenario.parameters[param.id].toString();
+    return param.options ? param.options.find(({ id }) => id === rawVal)!.label : rawVal;
+  }
+};
+
+const toggleOpen = (seriesId: string) => {
+  if (openedTimeSeriesAccordions.value.includes(seriesId)) {
+    openedTimeSeriesAccordions.value = openedTimeSeriesAccordions.value.filter(id => id !== seriesId);
+  } else {
+    openedTimeSeriesAccordions.value = [...openedTimeSeriesAccordions.value, seriesId];
+  }
+};
+
+const onMouseLeaveTimeSeries = () => {
+  setTimeout(() => {
+    hideTimeSeriesTooltips.value = true;
+  }, 500);
+};
+
+// Eagerly try to load the status and results, in case they are already available and can be used during server-side rendering.
+await appStore.loadScenarioStatus();
+if (appStore.currentScenario.status.data?.runSuccess) {
+  appStore.loadScenarioResult();
+}
+
+let statusInterval: NodeJS.Timeout;
+const loadScenarioStatus = () => {
+  appStore.loadScenarioStatus().then(() => {
+    if (appStore.currentScenario.status.data?.runSuccess) {
+      clearInterval(statusInterval);
+      appStore.loadScenarioResult();
+    }
+  });
 };
 
+watch(() => (Object.keys(appStore.timeSeriesData || {})), (seriesIds) => {
+  openedTimeSeriesAccordions.value = seriesIds;
+});
+
 onMounted(() => {
   appStore.globe.interactive = false;
+
+  if (!appStore.currentScenario.status.data?.done || appStore.currentScenario.result.data) {
+    statusInterval = setInterval(loadScenarioStatus, 200); // Poll for status every N ms
+    setTimeout(() => {
+      clearInterval(statusInterval); // Terminate polling for status after several seconds
+    }, 5000);
```

Comment:
This should probably result in some error being printed to screen about the model run timing out (if the interval hadn't already been cleared). Is 5s long enough

Reply:
I updated this behavior to be as you see in this screen recording:

[Screencast from 06-10-24 23:46:41.webm](https://github.com/user-attachments/assets/7e4939c0-f1a2-4202-a2dd-af138eefee62)

Namely, it gives up after 15 seconds, but warns that things are taking longer than expected at 5 seconds.

For future reference, you can configure Mockoon to induce this behaviour in the app (by mocking the R API scenario status endpoint to say 'pending' forever), using the below Mockoon config:

[mockoon.json](https://github.com/user-attachments/files/17271149/mockoon.json)